test/rebar_utils_SUITE.erl

```diff
@@ -322,6 +322,7 @@ is_list_of_strings(_Config) ->
     ?assert(rebar_utils:is_list_of_strings("foo") == false).
 
 url_append_path(_Config) ->
-    ?assertEqual({ok, "https://repo.hex.pm:443/repos/org"}, rebar_utils:url_append_path("https://repo.hex.pm", "/repos/org")),
-    ?assertEqual({ok, "https://repo.hex.pm:443/repos/org?foo=bar"}, rebar_utils:url_append_path("https://repo.hex.pm",
```

Comment:
I think this test case is wrong. `/repos/org?key=value` is not a path segment. The new module will correctly encode the question mark but previous string concatenation one would leave it unencoded. So the test was updated to use a more sensible path "extension".

Reply:
Agreed. Hopefully we just didn't have a thing that relied on that behaviour in a critical way.